haskell-src/Concordium/Genesis/Data/P10.hs

```diff
@@ -0,0 +1,173 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | This module defines the genesis data format for the 'P10' protocol version.
+module Concordium.Genesis.Data.P10 where
+
+import Data.Serialize
+import Data.Word
+
+import Concordium.Common.Version
+import qualified Concordium.Crypto.SHA256 as Hash
+import qualified Concordium.Genesis.Data.Base as Base
+import qualified Concordium.Genesis.Data.BaseV1 as BaseV1
+import Concordium.Types
+import Concordium.Types.Updates
+
+-- | Parameters data type for the 'P9' to 'P10' protocol update.
+--  This is provided as a parameter to the protocol update chain update instruction.
+data ProtocolUpdateData = ProtocolUpdateData
+    { -- | Access structure defining the keys and threshold for CreatePLT chain update.
+      updateCreatePLTAccessStructure :: !AccessStructure
+    }
+    deriving (Eq, Show)
+
+instance Serialize ProtocolUpdateData where
+    put ProtocolUpdateData{..} = put updateCreatePLTAccessStructure
+    get = do
+        updateCreatePLTAccessStructure <- get
+        return ProtocolUpdateData{..}
```

Comment:
This is not needed for the P9->P10 update - we already have the Create PLT access structure.